fashion_mnist/README.md

```diff
@@ -0,0 +1,394 @@
+    if epoch < 25:
+        return 1e-1 * args.base_lr
+    if epoch < 35:
+        return 1e-2 * args.base_lr
+    return 1e-3 * args.base_lr
+
+
+callbacks = [
+    ...
+    
+    keras.callbacks.LearningRateScheduler(lr_schedule),
+    ...
+```
+
+with:
+
+```python
+callbacks = [
+    ...
+
+    # Horovod: using `lr = 1.0 * hvd.size()` from the very beginning leads to worse final
+    # accuracy. Scale the learning rate `lr = 1.0` ---> `lr = 1.0 * hvd.size()` during
+    # the first five epochs. See https://arxiv.org/abs/1706.02677 for details.
+    hvd.callbacks.LearningRateWarmupCallback(warmup_epochs=args.warmup_epochs, verbose=verbose),
+
+    # Horovod: after the warmup reduce learning rate by 10 on the 15th, 25th and 35th epochs.
+    hvd.callbacks.LearningRateScheduleCallback(start_epoch=args.warmup_epochs, end_epoch=15, multiplier=1.),
+    hvd.callbacks.LearningRateScheduleCallback(start_epoch=15, end_epoch=25, multiplier=1e-1),
+    hvd.callbacks.LearningRateScheduleCallback(start_epoch=25, end_epoch=35, multiplier=1e-2),
+    hvd.callbacks.LearningRateScheduleCallback(start_epoch=35, multiplier=1e-3),
+
+    ...
+```
+
+![image](https://user-images.githubusercontent.com/16640218/53535420-98e3c180-3ab7-11e9-8780-9258081f66c5.png)
+
+Since we've added new `args.warmup_epochs` argument, we should register it:
+
+```python
+parser.add_argument('--warmup-epochs', type=float, default=5,
+                    help='number of warmup epochs')
+```
+
+![image](https://user-images.githubusercontent.com/16640218/53535703-9d5caa00-3ab8-11e9-85aa-0bc6f93f9826.png)
+
+### Save checkpoints & logs only of the first worker
+
+We don't want multiple workers to be overwriting same checkpoint files, since it could lead to corruption.
+
+Replace the following:
+
+```python
+callbacks = [
+    ...
+
+    keras.callbacks.ModelCheckpoint(args.checkpoint_format),
+    keras.callbacks.TensorBoard(args.log_dir)
+]
+```
+
+with:
+
+```python
+callbacks = [
+    ...
+]
+
+# Horovod: save checkpoints only on the first worker to prevent other workers from corrupting them.
+if hvd.rank() == 0:
+    callbacks.append(keras.callbacks.ModelCheckpoint(args.checkpoint_format))
+    callbacks.append(keras.callbacks.TensorBoard(args.log_dir))
+```
+
+![image](https://user-images.githubusercontent.com/16640218/53535616-4e167980-3ab8-11e9-82d9-82e431dfd621.png)
+
+### Modify training loop to execute fewer steps per epoch
+
+In order to speed up training, we will execute fewer steps of distributed training.  In order to keep the total number of examples processed during the training the same, we will do `num_steps / N` steps, where `num_steps` is the original number of steps, and `N` is the total number of workers.
+
+We will also speed-up validation by validating `3 * num_validation_steps / N` steps on each worker.  Multiplier **3** provides over-sampling of validation data helps to increase probability that every validation example will be evaluated.
```

Comment:
speed-up --> speed up
Multiplier --> The multiplier 
data helps --> data, which helps
increase --> increase the